libpldmresponder/platform.cpp

```diff
@@ -261,6 +222,89 @@ Response Handler::getPDR(const pldm_msg* request, size_t payloadLength)
     return response;
 }
 
+Response Handler::getPDR(const pldm_msg* request, size_t payloadLength)
+{
+    if (hostPDRHandler)
+    {
+        if (hostPDRHandler->isHostUp() && oemPlatformHandler != nullptr)
+        {
+            if (!oemPlatformHandler->getBMCState())
+            {
+                return ccOnlyResponse(request, PLDM_ERROR_NOT_READY);
+            }
+
+            // This is R/R case where host is UP, so defer the PDR generation
+            // on to a new thread & unblock the pldm daemon
+            auto expected = pldmRepoStatus::PLDM_REPO_NOT_BUILT;
+            bool notBuilt = repoCreated.compare_exchange_strong(
+                expected, pldmRepoStatus::PLDM_REPO_BUILDING_IN_PROGRESS);
+            if (notBuilt)
+            {
+                // The repository is not build, return NOT_READY and
+                // spawn a new thread to generate the PDR's and unblock
+                // the PLDM daemon
+                std::thread pdrGenThread(&Handler::_generatePDRs, this);
+
+                // detach the thread, so that the main thread wont be
+                // waiting for the response of the secondary thread
+                pdrGenThread.detach();
+
+                return ccOnlyResponse(request, PLDM_ERROR_NOT_READY);
+            }
+            else if (repoCreated.load() ==
+                     pldmRepoStatus::PLDM_REPO_BUILDING_IN_PROGRESS)
+            {
+                return ccOnlyResponse(request, PLDM_ERROR_NOT_READY);
+            }
+            else if (repoCreated.load() ==
+                     pldmRepoStatus::PLDM_REPO_ALREADY_BUILT)
+            {
+                if (dbusToPLDMEventHandler && pdrCreated)
+                {
+                    deferredGetPDREvent =
+                        std::make_unique<sdeventplus::source::Defer>(
+                            event,
+                            std::bind(std::mem_fn(
+                                          &pldm::responder::platform::Handler::
+                                              _processPostGetPDRActions),
+                                      this, std::placeholders::_1));
+                }
+                return sendPDRs(request, payloadLength);
+            }
```

Comment:
I think you need

```
} else {
    assert("Unreachable");
}
```
or something equivalent to make sure we have complete coverage of the enum.